lib/config/sfFactoryConfigHandler.class.php

```diff
@@ -41,7 +41,7 @@ public function execute($configFiles)
     $instances = array();
 
     // available list of factories
-    $factories = array('view_cache_manager', 'logger', 'i18n', 'controller', 'request', 'response', 'routing', 'storage', 'user', 'view_cache', 'mailer', 'service_container');
+    $factories = array('service_container', 'view_cache_manager', 'logger', 'i18n', 'controller', 'request', 'response', 'routing', 'storage', 'user', 'view_cache', 'mailer');
```

Comment:
The service container depends on `user` and `routing` (e.g; `sfContext::getServiceContainer()`).

For example try to get a `service` that depend on `user` inside the constructor of the `response` class.
You will caught that the `sfContext::getInstance()->getUser()` return `null`. 

---

Another topic but it would be good to provide at least `i18n` and `mailer` as service.